docs/source/extension/extension_multiple_ui.rst

```diff
@@ -0,0 +1,277 @@
+.. Copyright (c) Jupyter Development Team.
+.. Distributed under the terms of the Modified BSD License.
+
+.. _multiple_ui_extensions:
+
+Extensions Targeting Multiple Applications
+==========================================
```

Comment:
This is kind of long. I  think we can remove "Extensions" because this doc is in "Develop Extensions" chapter.

```suggestion
Targeting Multiple Applications
===============================
```

Do you feel strongly about keeping "Extensions" part in?

![image](https://github.com/jupyterlab/jupyterlab/assets/5832902/fc791395-8f5f-48f4-a02c-ae260fecec89)